cfme/control/explorer/alerts.py

```diff
@@ -124,6 +126,71 @@ def is_displayed(self):
         )
 
 
+class MonitorAlertsAllView(BaseLoggedInPage):
+
+    result_header = Text('h5.ng-binding')
+    alerts_list = AllAlertsList('list-view-pf-view ng-isolate-scope')
+    # Used for Ascending/Descending sort
+    sort_order = Text(".//button[./span[contains(@class,'sort-direction')]]")
+    # Used to select filter_by items like 'Name', 'Severity'
+    filter_by_dropdown = SelectorDropdown('uib-tooltip', 'Filter by')
+    # Used to select sort by options like 'Name', 'Number of Associated Plans'
+    sort_by_dropdown = SelectorDropdown('class','btn btn-default ng-binding dropdown-toggle')
+
+    @property
+    def in_monitor_alerts_all(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Monitor', 'Alerts', 'All Alerts']
+        )
+
+    @property
+    def is_displayed(self):
+        return self.in_monitor_alerts_all
+
+    @View.nested
+    class history_table(View):
+        """ history table only appears after clicking an item in the alerts list """
+        history_table = Table('.//table[contains(@class,"table table-striped")]')
```

Comment:
'table table-striped' are too generic, I think - are there more specific locators for the table within the markup, or is that it? If so let's open an RFE for some more specific attributes in the DOM and link the BZ here.